src/main/java/seedu/duke/Duke.java

```diff
@@ -0,0 +1,784 @@
+    public abstract boolean isExit();
+}
+
+class MarkCommand extends Command {
+    private int index;
+
+    public MarkCommand(int index) {
+        this.index = index;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        if (index < 0 || index >= tasks.getSize()) {
+            throw new DukeException("Invalid task number.");
+        }
+        Task task = tasks.getTask(index);
+        task.markAsDone();
+        ui.showMarkedTask(task);
+        storage.save(tasks);
+    }
+
+    @Override
+    public boolean isExit() {
+        return false;
+    }
+}
+
+class UnmarkCommand extends Command {
+    private int index;
+
+    public UnmarkCommand(int index) {
+        this.index = index;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        if (index < 0 || index >= tasks.getSize()) {
+            throw new DukeException("Invalid task number.");
+        }
+        Task task = tasks.getTask(index);
+        task.markAsNotDone();
+        ui.showUnmarkedTask(task);
+        storage.save(tasks);
+    }
+
+    @Override
+    public boolean isExit() {
+
+        return false;
+    }
+}
+
+class DeleteCommand extends Command {
+    private int index;
+
+    public DeleteCommand(int index) {
+
+        this.index = index;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        if (index < 0 || index >= tasks.getSize()) {
+            throw new DukeException("Invalid task number.");
+        }
+        Task task = tasks.removeTask(index);
+        ui.showDeletedTask(task, tasks.getSize());
+        storage.save(tasks);
+    }
+
+    @Override
+    public boolean isExit() {
+
+        return false;
+    }
+}
+
+class ExitCommand extends Command {
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        ui.showGoodbye();
+    }
+
+    @Override
+    public boolean isExit() {
+
+        return true; // Indicate that the application should exit
+    }
+}
+
+
+
+class ListCommand extends Command {
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        ui.showTaskList(tasks);
+    }
+    public boolean isExit() {
+
+        return false;
+    }
+}
+
+class AddTodoCommand extends Command {
+    private String description;
+
+    public AddTodoCommand(String description) {
+
+        this.description = description;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        Todo newTodo = new Todo(description);
+        tasks.addTask(newTodo);
+        ui.showTaskAdded(newTodo, tasks.getSize());
+        storage.save(tasks);
+    }
+
+    @Override
+    public boolean isExit() {
+
+        return false;
+    }
+}
+
+class AddDeadlineCommand extends Command {
+    private String description;
+    private LocalDateTime by;
```

Comment:
Should the variable name be more descriptive as it is used throughout the class? Ex. deadline, target_date, etc.